math/src/msm/pippenger.rs

```diff
@@ -0,0 +1,146 @@
+use crate::{cyclic_group::IsGroup, unsigned_integer::element::UnsignedInteger};
+
+/// This function computes the multiscalar multiplication (MSM).
+///
+/// Assume a group G of order r is given.
+/// Let `hidings = [g_1, ..., g_n]` be a tuple of group points in G and
+/// let `cs = [k_1, ..., k_n]` be a tuple of scalars in the Galois field GF(r).
+///
+/// Then, with additive notation, `msm(cs, hidings)` computes k_1 * g_1 + .... + k_n * g_n.
+///
+/// If `hidings` and `cs` are empty, then `msm` returns the zero element of the group.
+///
+/// Panics if `cs` and `hidings` have different lengths.
```

Comment:
I would recommend it doesn't panic, but if it does, then the `debug_assert_eq!` should be replaced by a regular `assert_eq!`.

Reply:
Changed it. Now it returns a `Result` that is later unwrapped by callers.